verify/verify.go

```diff
@@ -24,12 +23,24 @@ func (db *DB) Verify(s *data.Signed, role string, minVersion int) error {
 	if err := json.Unmarshal(s.Signed, sm); err != nil {
 		return err
 	}
-	if strings.ToLower(sm.Type) != strings.ToLower(role) {
-		return ErrWrongMetaType
+
+	if isTopLevelRole(role) {
+		// Top-level roles can only sign metadata of the same type (e.g. snapshot
+		// metadata must be signed by the snapshot role).
+		if sm.Type != role {
```

Comment:
Why no lowercasing of both sides here?

Reply:
good catch, covered here https://github.com/raphaelgavache/go-tuf/pull/5/commits/3078b69258457de0967f5dc136a38a8b1109b62c